### DISCUSSION TITLE
Component within repeater is not refreshing dynamically

### DISCUSSION THREAD

**Discussion by Elestara** [Budibase/budibase#14030] [Help] [answered]
Hello everyone, little confused here.

I have Postgres custom query table which includes custom checkbox system (data provider->repeater->form->checkbox, all with custom queries to manipulate)

And I also have custom filter made by option picker+option picker value filter on data provider (schema I picked here and there in discussions). Filter works perfect on my table, except for checkbox if its checked. 

My goal is to clear all checkboxes after "filtering"and table change. To do so I set "on change" for option picker to uncheck all checkboxes (via query) and refresh data provider. 
In result, in data table all fine and clear, but in app checkboxes remain checked/kinda mismatched even until I refresh page completely.
Update form field value for checkbox or clear form also not working to make it visually correct.

Is this a bug or am I missing somethig?

**Comment by mikesealey** [comment]
Hey @Elestara 

This may be a bug - I've raised a bug ticket for the engineering team to look into. https://github.com/Budibase/budibase/issues/14032

For the time being, maybe I could help you find a workaround. Could you share some more information about what it is you're hoping to achieve?

**Reply by Elestara** [reply]
Hey @mikesealey 

Thank you for quick response! I hope bug will be fixed soon.

As I mentioned, I have custom table with repeater, form and checkbox and option picker that filters my table on data provider level.
My main goal is to reset checkbox selection after table change. 
For example, user opens table, makes a selection and then decides to apply filter. Ideally all selected checkboxes must be blank after applying or after some other manipulations with table when necessary. 

Not sure how to achieve it in other way, because checkbox can be only manipulate strictly through form and on data level all works fine. But I will be glad to hear any suggestions)

**Comment by Elestara** [comment, marked as answer]
So, I finally managed it.
Initially, the problem was that to "imitate select function in custom query table" was real checkbox inside form (and inside the repeater) from data table and it was impossible to manipulate it outside form, as was mentioned in issue above (and buttons or filters placed inside repeater are kinda silly). I tried to make embed checkbox but my knowlege of coding is short here. 
So I came with conditional UI instead. Checkbox in data table is still used but only as value for states and conditions to show png checkbox and paint rows) I still need to find nicer PNG but it's really operational now and I can imitate table functions properly.

PS I realised that I can simply use icons instead of png
![Анимация](https://github.com/user-attachments/assets/d0f53dc0-9ce4-4ff3-bce1-376aab0c586c)
![image](https://github.com/user-attachments/assets/a85e9be5-8d67-4511-bdec-d63aa2cffd46)
